sdks/python/apache_beam/io/vcfio.py

```diff
@@ -0,0 +1,428 @@
+                                if PHASESET_FORMAT_KEY in sample.data._fields
+                                else None)
+        # Note: Call is considered phased if it contains the 'PS' key regardless
+        # of whether it uses '|'.
+        if phaseset_from_format or sample.phased:
+          call.phaseset = (str(phaseset_from_format) if phaseset_from_format
+                           else DEFAULT_PHASESET_VALUE)
+        for field in sample.data._fields:
+          # Genotype and phaseset (if present) are already included.
+          if field in (GENOTYPE_FORMAT_KEY, PHASESET_FORMAT_KEY):
+            continue
+          data = getattr(sample.data, field)
+          # Convert single values to a list for cases where the number of fields
+          # is unknown. This is to ensure consistent types across all records.
+          # Note: this is already done for INFO fields in PyVCF.
+          if (field in formats and
+              formats[field].num is None and
+              isinstance(data, (int, float, long, basestring, bool))):
+            data = [data]
+          call.info[field] = data
+        variant.calls.append(call)
+      return variant
+
+    def _get_field_count_as_string(self, field_count):
+      """Returns the string representation of field_count from PyVCF.
+
+      PyVCF converts field counts to an integer with some predefined constants
+      as specified in the vcf.parser.field_counts dict (e.g. 'A' is -1). This
+      method converts them back to their string representation to avoid having
+      direct dependency on the arbitrary PyVCF constants.
+      Args:
+        field_count (int): An integer representing the number of fields in INFO
+          as specified by PyVCF.
+      Returns:
+        A string representation of field_count (e.g. '-1' becomes 'A').
+      Raises:
+        ValueError: if the field_count is not valid.
+      """
+      if field_count is None:
+        return None
+      elif field_count >= 0:
+        return str(field_count)
+      field_count_to_string = {v: k for k, v in vcf.parser.field_counts.items()}
+      if field_count in field_count_to_string:
+        return field_count_to_string[field_count]
+      else:
+        raise ValueError('Invalid value for field_count: %d' % field_count)
+
+
+class ReadFromVcf(PTransform):
+  """A :class:`~apache_beam.transforms.ptransform.PTransform` for reading VCF
+  files.
+
+  Parses VCF files (version 4) using PyVCF library. If file_pattern specifies
+  multiple files, then the header from each file is used separately to parse
+  the content. However, the output will be a PCollection of
+  :class:`Variant` objects.
+  """
+
+  def __init__(
+      self,
+      file_pattern=None,
+      compression_type=CompressionTypes.AUTO,
+      validate=True,
+      **kwargs):
+    """Initialize the :class:`ReadFromVcf` transform.
+
+    Args:
+      file_pattern (str): The file path to read from as a local file path.
```

Comment:
<!--new_thread; commit:63511ddb8caf9e5c12888ef8a4d1071e0ea6c168; resolved:0-->
"The file path to read from either as a single file or a glob pattern." (documentation in textio.py is wrong as well, I'll fix that in a later PR).

Reply:
Done.